llvm/include/llvm/Frontend/OpenMP/OMP.td

```diff
@@ -1162,7 +1162,7 @@ def OMP_Workshare : Directive<"workshare"> {
   let category = CA_Executable;
 }
 def OMP_EndWorkshare : Directive<"end workshare"> {
-  let allowedClauses = [
+  let allowedOnceClauses = [
```

Comment:
nit: please could you add a test for this check as well. Something in `flang/test/` is okay

Otherwise this LGTM

Reply:
I have added a test here: https://github.com/llvm/llvm-project/pull/111358/files#diff-3b5218aa589f253f6fe3c0637625dfe6c38d9dc425d5290a626b2b6796f06db5R79-R82. Is it fine or should I move it to `flang/test`

Reply:
No issues, @tblah. Thanks for the review!